api/v1/handlers.go

```diff
@@ -1773,20 +1774,32 @@ type minerResp struct {
 // @Success      200  {object}  string
 // @Failure      400           {object}  util.HttpError
 // @Failure      500           {object}  util.HttpError
-// @Router       /public/miners [get]
+// @Param        params           path      bool  false   "set true to include on-chain miner info"
+// @Router       /public/miners/{lookup_chain} [get]
 func (s *apiV1) handleAdminGetMiners(c echo.Context) error {
+	ctx := context.TODO()
 	var miners []model.StorageMiner
 	if err := s.DB.Find(&miners).Error; err != nil {
 		return err
 	}
 
+	lookupChain, _ := strconv.ParseBool(c.Param("lookup_chain"))
+
 	out := make([]minerResp, len(miners))
+
 	for i, m := range miners {
 		out[i].Addr = m.Address.Addr
 		out[i].Suspended = m.Suspended
 		out[i].SuspendedReason = m.SuspendedReason
 		out[i].Name = m.Name
 		out[i].Version = m.Version
+
+		if lookupChain {
+			ci, err := s.minerManager.GetMinerChainInfo(ctx, m.Address.Addr)
```

Comment:
should we run in multiple threads to improve performance?

Reply:
We can run this in a go but we should cache the results as well. We can wrap this on the cacher (the server object has an instance of this) and return the result. This way, we only need to get the miner chain info of the specific address directly from chain once every hour.

Reply:
Cached in `ExtendedCacher`